src/main/java/abel/ArrayList.java

```diff
@@ -0,0 +1,86 @@
+package abel;
+
+import java.util.Arrays;
+
+/**
+ * Class that implements a simple Array List.
+ *
+ * @param <T> Generic Object.
+ */
+public class ArrayList<T> implements List {
+
+    public static final double GROW_RATE = 1.5;
+    private int size;
+    private Object[] array;
+    private static final int INITIAL_SIZE = 10;
+    private static final Object[] INITIAL_ARRAY = new Object[0];
+    private int currentArraySize;
+
+    /**
+     * Default Constructor.
+     */
+    public ArrayList() {
+        array = INITIAL_ARRAY;
```

Comment:
This is fine, I think it is a valid idea to have a constructor that initialize the internal array to 0 but don't forget to also give the client the ability to choose its own initial size by providing a new constructor